File: php/examples/simple_webserver.php
<?php
session_start();

if (!isset($_SESSION['count'])) $_SESSION['count'] = 0;
$_SESSION['count']++;

$count = $_SESSION['count'];
switch($count) {
case 1:
    $message = "Hohhoo";
    break;
case 2:
    $message = "Hioh";
    break;
case 3:
    $message = "HOhoKKK";
    break;
default:
    $message = "I've reached " . $count . " level";
}

echo json_encode(["Count" => $count, "Message" => $message]);

File: go/examples/simple_webserver.go
package main

import (
	"fmt"
	"net/http"
	"encoding/json"
)


type Counter int
type Response struct {
    Body string `json:"body"` //field tags
    Count int `json:"counter"`
}

func (ctr *Counter) ServeHTTP(w http.ResponseWriter, req *http.Request) {
	w.Header().Set("Content-Type", "application/json")

	message := ""
	*ctr++

	switch *ctr {
	case 1:
		message = "Hohhoo"
	case 2:
		message = "Hioh"
	case 3:
		message = "HOhoKKK"
	default:
		message = fmt.Sprint("I've reached ", *ctr, " level")
	}

	res := &Response { Body: message, Count: int(*ctr) }
	enc := json.NewEncoder(w)
	enc.Encode(res)
}



func main() {
	ctr := new(Counter)
	http.Handle("/counter", ctr)
	http.ListenAndServe(":8080", nil)
}
